### PR TITLE
Enhanced smoke test - check for missing results

### PR DESCRIPTION
**Pull request by crutchfield**
#  #

## 🗣 Description ##

Check if any missing test results are reported in HTML individual reports
Closing #281 

## 💭 Motivation and context ##

Verifies all controls in baseline documents have an explicit implementation even if "not implemented" implementation. 

## 🧪 Testing ##

Execute smoke test from action menu.  Verify test cases fail only for missing results (for now in Emerald).  Smoke test will need to pass before promoting Emerald branch to main. 

## 📷 Screenshots (if appropriate) ##
image

## ✅ Pre-approval checklist ##

- [X] This PR has an informative and human-readable title.
- [X] Changes are limited to a single goal - *eschew scope creep!*
- [X] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [X] All relevant type-of-change labels have been added.
- [X] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [X] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [X] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [X] Tests have been added and/or modified to cover the changes in this PR.
- [ X] All new and existing tests pass (except as annotated above).

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

**Comment by Dylan-MITRE** [comment]
Hi Crutch there are some check were not successful, I am wondering if you are working on it. And do you want me to review it now or after those error got resolved?

**Comment by crutchfield** [comment]
> Hi Crutch there are some check were not successful, I am wondering if you are working on it. And do you want me to review it now or after those error got resolved?

We are expecting some errors for now - untill rego is fully aligned with updated secure baseline.  You can start review.

File: Testing/Unit/PowerShell/CreateReport/New-Report.Tests.ps1 [hunk 1]
```diff
@@ -3,7 +3,8 @@ Import-Module (Join-Path -Path $PSScriptRoot -ChildPath '../../../../PowerShell/
 InModuleScope CreateReport {
     Describe -Tag CreateReport -Name 'New-Report' {
         BeforeAll {
-            Mock -CommandName Write-Error {}
+            Mock -CommandName Write-Warning {}
+
```
Comment (Dylan-MITRE) [inline Testing/Unit/PowerShell/CreateReport/New-Report.Tests.ps1:7]: Is Line 7 creation on purpose?

Comment (crutchfield) [reply Testing/Unit/PowerShell/CreateReport/New-Report.Tests.ps1:7]: If I understand the question, "Do I intend to have a blank line"?
I can remove if desired.

Comment (Dylan-MITRE) [reply Testing/Unit/PowerShell/CreateReport/New-Report.Tests.ps1:7]: Oh i don't have a preference, just wanted to make sure its not an accidental blank line.

**Review by adhilto — APPROVED** [review]
Minus the merge conflicts, looks good.

File: Testing/Unit/PowerShell/CreateReport/New-Report.Tests.ps1 [hunk 2]
```diff
@@ -41,22 +42,22 @@ InModuleScope CreateReport {
...
         It 'Creates a report for <Product>' -ForEach @(
-            @{Product = 'aad'; ErrorCount = 1},
-            @{Product = 'defender'; ErrorCount = 3},
-            @{Product = 'exo'; ErrorCount = 2},
-            @{Product = 'onedrive'; ErrorCount = 8},
-            @{Product = 'powerplatform'; ErrorCount = 0},
-            @{Product = 'sharepoint'; ErrorCount = 5},
-            @{Product = 'teams'; ErrorCount = 5}
+            @{Product = 'aad'; WarningCount = 1},
+            @{Product = 'defender'; WarningCount = 3},
```
Comment (crutchfield) [inline Testing/Unit/PowerShell/CreateReport/New-Report.Tests.ps1:46]:
```suggestion
            @{Product = 'defender'; WarningCount = 10},
```

File: Testing/Unit/PowerShell/CreateReport/New-Report.Tests.ps1 [hunk 3]
```diff
@@ -41,22 +42,22 @@ InModuleScope CreateReport {
...
+            @{Product = 'exo'; WarningCount = 2},
+            @{Product = 'onedrive'; WarningCount = 8},
+            @{Product = 'powerplatform'; WarningCount = 0},
+            @{Product = 'sharepoint'; WarningCount = 3},
```
Comment (crutchfield) [inline Testing/Unit/PowerShell/CreateReport/New-Report.Tests.ps1:50]:
```suggestion
            @{Product = 'sharepoint'; WarningCount = 5},
```

**Pull request merged**